test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -234,9 +239,109 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 
 			ginkgo.By("checking that the finalizer is removed when the Pod is deleted early")
 			gomega.Expect(k8sClient.Delete(ctx, createdPod)).Should(gomega.Succeed())
-			gomega.Eventually(func(g gomega.Gomega) bool {
-				return apierrors.IsNotFound(k8sClient.Get(ctx, lookupKey, createdPod))
-			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func(g gomega.Gomega) error {
+				return k8sClient.Get(ctx, lookupKey, createdPod)
+			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
+		})
+
+		ginkgo.It("Should reconcile the pod group with the queue name", func() {
+			ginkgo.By("Creating pods with queue name")
+			pod1 := testingpod.MakePod("test-pod1", ns.Name).
+				Group("test-group").
+				GroupTotalCount("2").
+				Queue("test-queue").
+				Obj()
+			pod2 := testingpod.MakePod("test-pod2", ns.Name).
+				Group("test-group").
+				GroupTotalCount("2").
+				Image("test-image", nil).
+				Queue("test-queue").
+				Obj()
+			pod1LookupKey := client.ObjectKeyFromObject(pod1)
+			pod2LookupKey := client.ObjectKeyFromObject(pod2)
+
+			gomega.Expect(k8sClient.Create(ctx, pod1)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, pod2)).Should(gomega.Succeed())
+
+			ginkgo.By("checking that workload is created for the pod group with the queue name")
+			wlLookupKey := types.NamespacedName{
+				Namespace: pod1.Namespace,
+				Name:      "test-group",
+			}
+			createdWorkload := &kueue.Workload{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			gomega.Expect(createdWorkload.Spec.PodSets).To(gomega.HaveLen(2))
+			gomega.Expect(createdWorkload.Spec.PodSets[0].Count).To(gomega.Equal(int32(1)))
+			gomega.Expect(createdWorkload.Spec.PodSets[0].Name).To(gomega.Equal("120fa2c0"))
+			gomega.Expect(createdWorkload.Spec.PodSets[1].Count).To(gomega.Equal(int32(1)))
+			gomega.Expect(createdWorkload.Spec.PodSets[1].Name).To(gomega.Equal("542d5455"))
+			gomega.Expect(createdWorkload.Spec.QueueName).To(gomega.Equal("test-queue"), "The Workload should have .spec.queueName set")
+
+			createdPod := &corev1.Pod{}
+			ginkgo.By("checking that all pods in group are unsuspended when workload is admitted", func() {
+				admission := testing.MakeAdmission(clusterQueue.Name, "120fa2c0", "542d5455").
+					Assignment(corev1.ResourceCPU, "default", "1").
+					AssignmentPodCount(2).
+					Obj()
+				gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+
+				gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
```

Comment:
worth adding a helper function `ExpectPodsToBeActiveWithNodeSelector`, where the node selector can be empty.

Reply:
done